litestar/plugins/flash.py

```diff
@@ -57,26 +61,19 @@ def on_app_init(self, app_config: AppConfig) -> AppConfig:
         return app_config
 
 
-def flash(connection: ASGIConnection, message: str, category: str) -> None:
-    """Add a flash message to the request scope.
-
-    Args:
-        connection: The connection instance.
-        message: The message to flash.
-        category: The category of the message.
-    """
-    scope_state = ScopeState.from_scope(connection.scope)
-    scope_state.flash_messages.append({"message": message, "category": category})
+def flash(
+    request: Request,
+    message: Any,
+    category: str,
+) -> None:
+    if "_messages" not in request.session:
+        request.session["_messages"] = []
+    request.session["_messages"].append({"message": message, "category": category})
 
 
 def get_flashes(context: Mapping[str, Any]) -> Any:
-    """Get flash messages from the request scope, if any.
-
-    Args:
-        context: The context dictionary.
-
-    Returns:
-        The flash messages, if any.
-    """
-    scope_state = ScopeState.from_scope(_get_request_from_context(context).scope)
-    return scope_state.flash_messages
+    return (
```

Comment:
just a nit, can this be simplified to `_get_request_from_context(context).session.pop("_messages", [])`?